Sources/TensorFlow/Optimizer.swift

```diff
@@ -201,8 +201,7 @@ public class RMSProp<Model: Layer>: Optimizer
 ///
 /// An optimizer that implements stochastic gradient descent, with support for momentum, learning
 /// rate decay, and Nesterov momentum.
-public class SGD<Model: Layer>: Optimizer
-    where Model.AllDifferentiableVariables == Model.TangentVector {
+public class SGD<Model: Layer>: Optimizer where Model.TangentVector: VectorProtocol {
```

Comment:
Can we also relax this to conform to `Differentiable` instead of `Layer` so that optimizers can be used even without necessarily using layers?

Reply:
Yes, absolutely!

Reply:
Great, thanks! :)

Reply:
Done.